src/paho/mqtt/subscribe.py

```diff
@@ -180,18 +198,27 @@ def callback(callback, topics, qos=0, userdata=None, hostname="localhost",
             client.tls_set_context(tls)
 
     client.connect(hostname, port, keepalive)
-    client.loop_forever()
+
+    if timeout == None:
+        client.loop_forever()
+    else:
+        lock.acquire()
+        client.loop_start()
+        lock.acquire(timeout=timeout)
```

Comment:
Since this is a one-on-one communication I believe that a Lock is actually modelling the relationship better. Do you have any additional use in mind e.g., an observer/debugger watching these events? Otherwise I would tend to see Lock as better.

Reply:
I feel that an Event is even better at modelling this: the finishing processing is an event and we wait for that event.

With the lock, we have two consecutive call to acquire within the same thread isn't usual way to work with a lock.